### PR TITLE
 Added the specified requirements. closes #80

### LINKED ISSUE #80: Random Resource distribution over the ResourceMap

### PR DESCRIPTION
**Pull request by Ativelox**
- Redid the resources class to hold a width and height, since it is represented as an ellipse (an ellipse is defined by a rectangle)

- Added a DrawEllipse method in the Primitives2D library with a cache to reduce calculation effort.

- Assigned colors to the resource types. These are arbitrary and need some reworking when decided.

- Added a random resource distribution and showcased it on the gamescreen.

### CODE REVIEW

File: Singularity/Singularity/Libraries/Primitives2D.cs [hunk 1]
```diff
@@ -19,8 +20,9 @@ public static class Primitives2D
 		#region Private Members
 
 		private static readonly Dictionary<String, List<Vector2>> sCircleCache = new Dictionary<string, List<Vector2>>();
-		//private static readonly Dictionary<String, List<Vector2>> arcCache = new Dictionary<string, List<Vector2>>();
-		private static Texture2D sPixel;
+	    private static readonly Dictionary<Rectangle, List<Vector2>> sEllipseCache = new Dictionary<Rectangle, List<Vector2>>();
```
Comment (fkarg) [inline Singularity/Singularity/Libraries/Primitives2D.cs]: how come the indentation is differing?

File: Singularity/Singularity/Libraries/Primitives2D.cs [hunk 2]
```diff
@@ -90,6 +94,70 @@ private static List<Vector2> CreateCircle(double radius, int sides)
...
 
+	    private static List<Vector2> CreateEllipse(Rectangle rect)
+	    {
+	        if (sEllipseCache.ContainsKey(rect))
+	        {
+	            return sEllipseCache[rect];
+	        }
+
+            var vectors = new List<Vector2>();
+
```
Comment (fkarg) [inline Singularity/Singularity/Libraries/Primitives2D.cs]: differing indentation?
(line before and after)

File: Singularity/Singularity/Libraries/Primitives2D.cs [hunk 3]
```diff
@@ -90,6 +94,70 @@ private static List<Vector2> CreateCircle(double radius, int sides)
...
+	                }
+                    //System.Diagnostics.Debug.Write(x + ", "  + y);
+	                vectors.Add(new Vector2(x, y));
+
+                }
+
+
+            }
+	        for (var x = rect.X + rect.Width; x >= rect.X; x--)
```
Comment (fkarg) [inline Singularity/Singularity/Libraries/Primitives2D.cs]: here as well?

File: Singularity/Singularity/Libraries/Primitives2D.cs [hunk 4]
```diff
@@ -441,83 +509,112 @@ public static void PutPixel(this SpriteBatch spriteBatch, Vector2 position, Colo
...
+	    /// <param name="spriteBatch">The destination drawing surface</param>
+	    /// <param name="rect">The rectangle which desribes the ellipse</param>
+	    /// <param name="color">The color of the ellipse</param>
+        /// <param name="thickness">The thickness of the lines used</param>
+        public static void DrawEllipse(this SpriteBatch spriteBatch, Rectangle rect, Color color, float thickness, float layer)
 		{
-			DrawPoints(spriteBatch, center, CreateCircle(radius, sides), color, thickness);
-		}
+		    DrawPoints(spriteBatch, Vector2.Zero, CreateEllipse(rect), color, thickness, layer);
+        }
```
Comment (fkarg) [inline Singularity/Singularity/Libraries/Primitives2D.cs]: what happened here?

**Comment by Ativelox** [comment]
Hm strange, the identation actually looks fine on my end, maybe my VS made some tab/space issues, which get displayed differently on github?

**Comment by fkarg** [comment]
dunno, if you say it's looking good on your end then I'll trust you there

**Comment by Ativelox** [comment]
I'm going to look into it a bit further on my end

**Comment by fkarg** [comment]
well in that case go ahead :)

**Pull request merged**